src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -624,6 +634,14 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 				}
 
 				DispatchQueue.main.async {
+					if let recycleBinItemToRestore = self.model.recycleBinItemToRestore(for: testRegistrationInformation) {
+						self.dismiss {
+							self.showTestRestoredFromBinAlert(recycleBinItem: recycleBinItemToRestore)
```

Comment:
do we need here a [weak self]?

Reply:
Since the exposure submission coordinator is a bit special in the way that the navigation controller holds the strong reference to the coordinator, it doesn't work with a weak self here. The closure needs to hold a strong self reference so it continues to exist until the alert comes up. But in general weak self is never required for dismiss as the dismiss closure is not stored anywhere (at least not long-term) and therefore no reference cycle can occur